packages/react-native-tab-view/src/__tests__/index.test.tsx

```diff
@@ -49,3 +102,71 @@ test('switches tabs when the tab header for another route is pressed', () => {
   fireEvent.press(getByLabelText('Second'), {} as GestureResponderEvent);
   expect(getByTestId('route2')).toBeTruthy();
 });
+
+test('passes onPageSelected prop from TabView to Pager', () => {
+  const pagerMock = require('../Pager');
+  render(<ComponentWithTabView />);
+
+  // Verify onPageSelected was passed to the Pager
+  expect(pagerMock.__mockOnPageSelected).toBeDefined();
+});
+
+test('calls onPageSelected when a tab is selected', () => {
+  // Reset the mocks before this test
+  require('../Pager').__resetMocks();
+
+  const { getByLabelText } = render(<ComponentWithTabView />);
+
+  // Reset the onPageSelectedMock as well
+  onPageSelectedMock.mockReset();
+
+  // The initial render should not trigger onPageSelected
+  expect(onPageSelectedMock).not.toHaveBeenCalled();
+
+  // Simulate selecting the second tab
+  fireEvent.press(getByLabelText('Second'), {} as GestureResponderEvent);
+
+  // Verify onPageSelected was called with the correct event
+  expect(onPageSelectedMock).toHaveBeenCalledTimes(1);
+  expect(
+    (
+      onPageSelectedMock.mock.calls[0][0] as {
+        nativeEvent: { position: number };
+      }
+    ).nativeEvent
+  ).toHaveProperty('position', 1);
```

Comment:
Use `toMatchObject` & `expect.objectContaining` with `toHaveBeenCalledWith`  to check specific properties instead of accessing `.mock` property directly